common/statistics/stateStatistics_test.go

```diff
@@ -0,0 +1,200 @@
+package statistics
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewStateStatistics_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	ss := NewStateStatistics()
+
+	assert.False(t, check.IfNil(ss))
+}
+
+func TestStateStatistics_Processing(t *testing.T) {
+	t.Parallel()
+
+	t.Run("trie operations", func(t *testing.T) {
+		t.Parallel()
+
+		ss := NewStateStatistics()
+
+		assert.Equal(t, uint64(0), ss.Trie())
+
+		ss.IncrTrie()
+		ss.IncrTrie()
+		assert.Equal(t, uint64(2), ss.Trie())
+
+		ss.IncrTrie()
+		assert.Equal(t, uint64(3), ss.Trie())
+
+		ss.Reset()
+		assert.Equal(t, uint64(0), ss.Trie())
+	})
+
+	t.Run("persister operations", func(t *testing.T) {
+		t.Parallel()
+
+		ss := NewStateStatistics()
+
+		epoch := uint32(1)
+
+		assert.Equal(t, uint64(0), ss.Persister(epoch))
+
+		ss.IncrPersister(epoch)
+		ss.IncrPersister(epoch)
+		assert.Equal(t, uint64(2), ss.Persister(epoch))
+
+		ss.IncrPersister(epoch)
+		assert.Equal(t, uint64(3), ss.Persister(epoch))
+
+		ss.Reset()
+		assert.Equal(t, uint64(0), ss.Persister(epoch))
+	})
+
+	t.Run("cache operations", func(t *testing.T) {
+		t.Parallel()
+
+		ss := NewStateStatistics()
+
+		assert.Equal(t, uint64(0), ss.Cache())
+
+		ss.IncrCache()
+		ss.IncrCache()
+		assert.Equal(t, uint64(2), ss.Cache())
+
+		ss.IncrCache()
+		assert.Equal(t, uint64(3), ss.Cache())
+
+		ss.Reset()
+		assert.Equal(t, uint64(0), ss.Cache())
+	})
+}
+
+func TestStateStatistics_Snapshot(t *testing.T) {
+	t.Parallel()
+
+	t.Run("persister operations", func(t *testing.T) {
+		t.Parallel()
+
+		ss := NewStateStatistics()
+
+		epoch := uint32(1)
+
+		assert.Equal(t, uint64(0), ss.SnapshotPersister(epoch))
+
+		ss.IncrSnapshotPersister(epoch)
+		ss.IncrSnapshotPersister(epoch)
+		assert.Equal(t, uint64(2), ss.SnapshotPersister(epoch))
+
+		ss.IncrSnapshotPersister(epoch)
+		assert.Equal(t, uint64(3), ss.SnapshotPersister(epoch))
+
+		ss.ResetSnapshot()
+		assert.Equal(t, uint64(0), ss.SnapshotPersister(epoch))
+	})
+
+	t.Run("cache operations", func(t *testing.T) {
+		t.Parallel()
+
+		ss := NewStateStatistics()
+
+		assert.Equal(t, uint64(0), ss.Cache())
+
+		ss.IncrSnapshotCache()
+		ss.IncrSnapshotCache()
+		assert.Equal(t, uint64(2), ss.SnapshotCache())
+
+		ss.IncrSnapshotCache()
+		assert.Equal(t, uint64(3), ss.SnapshotCache())
+
+		ss.ResetSnapshot()
+		assert.Equal(t, uint64(0), ss.SnapshotCache())
+	})
+}
+
+func TestStateStatistics_Sync(t *testing.T) {
+	t.Parallel()
+
+	ss := NewStateStatistics()
+
+	epoch := uint32(1)
+
+	assert.Equal(t, uint64(0), ss.SyncCache())
+	assert.Equal(t, uint64(0), ss.SyncPersister(epoch))
+
+	ss.IncrPersister(epoch)
+	ss.IncrPersister(epoch)
+
+	ss.IncrCache()
+	ss.IncrCache()
+
+	ss.ResetSync()
+
+	assert.Equal(t, uint64(2), ss.SyncPersister(epoch))
```

Comment:
You increased normal persister, but sync persister also increased. Something's not right. Also, reset sync did nothing.

Reply:
same as for the above comment, that how i thought to implemented, ResetSync will set sync persister to normal persister, and SyncStats will get the difference